xrpl/asyncio/clients/client.py

```diff
@@ -0,0 +1,32 @@
+"""Interface for all network clients to follow."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from xrpl.models.requests.request import Request
+from xrpl.models.response import Response
+
+
+class Client(ABC):
+    """Interface for all network clients to follow."""
+
+    url: str
+
+    @abstractmethod
+    def __init__(self: Client) -> None:  # noqa: D107
+        raise NotImplementedError(f"{self.__class__.__name__} is abstract")
+
+    @abstractmethod
+    async def request_impl(self: Client, request: Request) -> Response:
```

Comment:
Why not just call this `request` or `request_async`? The name `request_impl` is very confusing to me

Reply:
`impl` == `implementation`, which is what this actually is. we have to have our implementations be async now as a result of this, but we still want to offer a sync `request` method for `SyncClient` and an async `request` method for `AsyncClient`